src/common/snippets/include/snippets/runtime_configurator.hpp

```diff
@@ -109,11 +116,21 @@ class RuntimeConfigurator {
      * @param linear_ir LinearIR
      */
     virtual void init_tensor_rank(const lowered::LinearIRCPtr& linear_ir) const;
+
+    struct UnifiedLoopInfoRtParams {
+        size_t work_amount = 0;
+        std::vector<int64_t> ptr_increments;
+        std::vector<int64_t> finalization_offsets;
+    };
+    static UnifiedLoopInfoRtParams compute_runtime_params(const lowered::UnifiedLoopInfoPtr& unified_loop_info);
+    using LoopInfoRuntimeParamsMap = std::unordered_map<lowered::UnifiedLoopInfoPtr, UnifiedLoopInfoRtParams>;
     /**
      * @brief Update Loop informations in LinearIR: Unified and ExpandedLoopInfo
      * @param linear_ir LinearIR
+     * @param initializated_info_map Reference on a map [LoopInfo->RuntimeParams].
+     * Can be used to pass in the method loop infos which were already initialized, e.g. by parallel domain optimization
      */
-    void update_loop_info(const lowered::LinearIRCPtr& linear_ir) const;
+    void update_loop_info(const lowered::LinearIRCPtr& linear_ir, LoopInfoRuntimeParamsMap& initializated_info_map) const;
```

Comment:
Please take a look at the description:
```
    /**
     * @brief Update Loop informations in LinearIR: Unified and ExpandedLoopInfo
     * @param linear_ir LinearIR
     * @param initializated_info_map Reference on a map [LoopInfo->RuntimeParams].
     * Can be used to pass in the method loop infos which were already initialized, e.g. by parallel domain optimization
     */
    void update_loop_info(const lowered::LinearIRCPtr& linear_ir, LoopInfoRuntimeParamsMap& initializated_info_map) const;
```

I tried to point out there that user can pass a map with already initialized infos => it is valid to pass non-empty `initializated_info_map`. Please let me know if the description is not clear enough -- I will try to reformulate it

Reply:
After reviewing this one and a few subsequent PRs, I realized that probably these kind of interfaces with a map reference as an argument is an inevitable and a lesser evil at this point. So let's leave it as is.
I think that the real problem is how the loop_infos' are stored, especially in the case splitted loops. To make these kind of updates more transparent, we need to reorganize loop_infos' connectivity on the design level.
Please remind me about this on the meeting, I'll share some more details about this. It would be interesting to know your opinion.